src/method.rs

```diff
@@ -207,8 +207,9 @@ impl Method {
     /// more words.
     pub fn is_idempotent(&self) -> bool {
         match self.0 {
-            Put | Delete | _ if self.is_safe() => true,
-            _ => false
+            Put | Delete => true,
+            _ if self.is_safe() => true,
```

Comment:
This can just be `m => m.is_safe()`

Reply:
The `is_safe` method is only implemented on `Method` not on `Inner` so changing to this gives method not found error.

Reply:
`_ => self.is_safe()` then.